executor/index_lookup_hash_join.go

```diff
@@ -556,10 +570,11 @@ func (iw *indexHashJoinInnerWorker) getMatchedOuterRows(innerRow chunk.Row, task
 		return nil, nil, nil
 	}
 	matchedRows = make([]chunk.Row, 0, len(iw.matchedOuterPtrs))
-	matchedRowIdx = make([]int, 0, len(iw.matchedOuterPtrs))
+	matchedRowPtr = make([]chunk.RowPtr, 0, len(iw.matchedOuterPtrs))
 	for _, ptr := range iw.matchedOuterPtrs {
-		rowIdx := int(ptr.RowIdx)
-		outerRow := task.outerResult.GetRow(rowIdx)
+		chkIdx, rowIdx := int(ptr.ChkIdx), int(ptr.RowIdx)
+		chk := task.outerResult.GetChunk(chkIdx)
+		outerRow := chk.GetRow(rowIdx)
```

Comment:
Seems these two lines can be replaced by `outerRow = task.outerResult.GetRow(ptr)`.